services/fdb/client/client.go

```diff
@@ -298,36 +311,83 @@ func (r *fdbCLICmd) SetFlags(f *flag.FlagSet) {
 	})
 }
 
-// Execute acts like the top level metadataCmd does and delegates actual execution to the parsed sub-command.
+// Execute acts like the top level fdbCmd does and delegates actual execution to the parsed sub-command.
 // It does pass along the top level request since any top level flags need to be filled in there and available
 // for the final request. Sub commands will implement the specific oneof Command and any command specific flags.
 func (r *fdbCLICmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	c := setupFDBCLI(f)
+	// If it's just asking for flags or commands run those and exit. Help is special since we have that
+	// as its own subcommand which executes remotely.
+	if f.Arg(0) == "flags" || f.Arg(0) == "commands" || f.Arg(0) == "help" {
+		return c.Execute(ctx, args...)
+	}
+
 	// We allow --exec and also just a trailing command/args to work. The former is to mimic how people are used
 	// to doing this from the existing CLI so migration is simpler.
-	if f.NArg() < 1 {
-		if r.exec == "" {
-			fmt.Fprintln(os.Stderr, "Must specify a command after all options or with --exec")
+	if f.NArg() > 0 {
+		if r.exec != "" {
+			fmt.Fprintln(os.Stderr, "Must specify a command after all options or with --exec but not both")
 			return subcommands.ExitFailure
 		}
+	}
+	if r.exec != "" {
 		// We want this to still use the parser so move the string from --exec in as Args to the flagset
 		f = flag.NewFlagSet("", flag.ContinueOnError)
 		args := strings.Fields(r.exec)
 		f.Parse(args)
 	}
-	c := setupFDBCLI(f)
+
+	c = setupFDBCLI(f)
+
 	args = append(args, r.req)
-	return c.Execute(ctx, args...)
+	var cmd []string
+
+	// Join everything back into one string and then resplit. So a command of fdbcli "kill; kill X" will
+	// parse through here one by one.
+	t := strings.Join(f.Args(), " ")
+	for _, a := range strings.Split(t, " ") {
```

Comment:
could you split on ";" and strings.Trim() the result of leading/trailing spaces?

Reply:
Yes, it does simplify this a lot in the end